### PR TITLE
Go support

### PR DESCRIPTION
**Pull request by egiptipavel**
## Description

I added Go tests generation.

### UI

Module `utbot-intellij-go` contains language-specific implementations of LanguageAssistant, DialogWrapper and other classes.

### CLI

Module `utbot-go-cli` contains command line generation and run tests interface for Go. 

### Manual tests

```Go
// DivOrPanic divides x by y or panics if y is 0
func DivOrPanic(x int, y int) int {
    if y == 0 {
        panic("div by 0")
    }
    return x / y
}
```

## Self-check list

- [x] I've set the proper **labels** for my PR (at least, for category and component).
- [x] PR **title** and **description** are clear and intelligible.
- [ ] I've added enough **comments** to my code, particularly in hard-to-understand areas.
- [ ] The functionality I've repaired, changed or added is covered with **automated tests**.
- [x] **Manual tests** have been provided optionally.
- [ ] The **documentation** for the functionality I've been working on is up-to-date.

### CODE REVIEW

File: utbot-go/src/main/kotlin/org/utbot/go/framework/api/go/GoApi.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * Parent class for all Go models.
+ *
+ * To see its children check GoUtModelsApi.kt at org.utbot.go.api.
+ */
+abstract class GoUtModel(
+    override val classId: GoTypeId,
+) : UtModel(classId) {
+    open fun getRequiredImports(): Set<String> = emptySet()
+    abstract fun canNotBeEqual(): Boolean
```
Comment (Markoutte) [inline utbot-go/src/main/kotlin/org/utbot/go/framework/api/go/GoApi.kt:54]: It is hard to understand code with negative assertion. `canBeEqual` is easier to understand.

**Comment by Markoutte** [comment]
Please, use [Mermaid](https://github.com/mermaid-js/mermaid#readme) to create document diagrams instead of binary images with diagrams (like `utbot-go/docs/diagrams/how-it-works.png` is used)

File: utbot-go/docs/DEVELOPERS_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+In the previous step, UTBot Go generated the values that the functions need to be tested with &mdash; now the task is to
+do this. Namely, execute the functions with the values generated for them and save the result.
+
+Essentially, the target function, the values generated for it, and the result of its execution form a test case. In
+fact, that is why this stage ends the global process of generating test cases.
+
+### Test code generation
+
+Finally, the last stage: the test cases are ready, UTBot Go needs only to generate code for them. Nothing terrible
```
Comment (Markoutte) [inline utbot-go/docs/DEVELOPERS_GUIDE.md:43]: > Nothing terrible happens here

Sounds not very good for documentation.

File: utbot-go/docs/DEVELOPERS_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+happens here, but one need to carefully consider the features of the Go language (for example, the necessity to cast
+constants to the desired type, oh).
+
+_That's how the world (UTBot Go) works!_
+
+## How to test UTBot Go
+
+_**TODO:**_ Gradle `runIde` task and building CLI application JAR locally.
```
Comment (Markoutte) [inline utbot-go/docs/DEVELOPERS_GUIDE.md:51]: To build CLI version the `build` on `utbot-cli-go` should be called

File: utbot-go/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+val apacheCommonsTextVersion: String? by rootProject
+val jacksonVersion: String? by rootProject
+val ideType: String? by rootProject
+val pythonCommunityPluginVersion: String? by rootProject
+val pythonUltimatePluginVersion: String? by rootProject
+
+tasks {
+    compileKotlin {
+        kotlinOptions {
+            jvmTarget = "11"
```
Comment (Markoutte) [inline utbot-go/build.gradle.kts:12]: It is preferrable to use 8 Java API for all modules except IntelliJ plugin and CLI

File: utbot-go/src/main/kotlin/org/utbot/go/api/GoTypesApi.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    override fun getRelativeName(packageName: String): String = "[$length]${elementTypeId.getRelativeName(packageName)}"
+}
+
+class GoInterfaceTypeId(
+    name: String,
+    implementsError: Boolean
+) : GoTypeId(name, implementsError = implementsError) {
+    override fun getRelativeName(packageName: String): String {
+        TODO("Not yet implemented")
```
Comment (Markoutte) [inline utbot-go/src/main/kotlin/org/utbot/go/api/GoTypesApi.kt:48]: Suspicious TODO

File: utbot-go/src/main/kotlin/org/utbot/go/framework/api/go/GoApi.kt [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import org.utbot.framework.plugin.api.*
+
+/**
+ * Parent class for all Go types for compatibility with UTBot framework.
+ *
+ * To see its children check GoTypesApi.kt at org.utbot.go.api.
+ */
+abstract class GoTypeId(
+    name: String, elementClassId: GoTypeId? = null, val implementsError: Boolean = false
+) : ClassId(name, elementClassId) {
```
Comment (Markoutte) [inline utbot-go/src/main/kotlin/org/utbot/go/framework/api/go/GoApi.kt:12]: We need to rid off all dependencies of core-api, like ClassId

Comment (tochilinak) [reply utbot-go/src/main/kotlin/org/utbot/go/framework/api/go/GoApi.kt:12]: We all have ClassId in dependencies (including Python, even though we have our own types). It is needed to use code generation, maybe some other places too.

**Review by Markoutte — APPROVED** [review]

**Pull request merged**